src/compute.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2018-2020, Michael P. Howard
+# Copyright (c) 2021-2024, Auburn University
+# Part of azplugins, released under the BSD 3-Clause License.
+
+"""Flow profiles."""
+
+import numpy
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.logging import log
+from hoomd.operation import Compute
+
+
+class FlowProfile(Compute):
```

Comment:
I believe this class would need to inherit from `Action` if we want to be able to attach it to the simulation, e.g., as a `CustomWriter`. However, we could also force the user to manually call it in a for loop of `run()` because HOOMD 4 does not have a concept of a callback or an "analyzer". @astatt what do you think?